File: CalcuFunction.py
import importlib
import os
import sys

class CalcuFunction:
    _extension_dir = os.path.dirname(os.path.abspath(__file__))

    # ----------------
    # Core math
    # ----------------
    @staticmethod
    def add(a, b): return a + b

    @staticmethod
    def sub(a, b): return a - b

    @staticmethod
    def product(a, b): return a * b

    @staticmethod
    def div(a, b): return a / b

    @staticmethod
    def exp(a, b): return a ** b

    @staticmethod
    def floor(a, b): return a // b

    # ----------------
    # CLI Calculator
    # ----------------
    @staticmethod
    def calculator():
        print("CalcuFunction Calculator")

        op = input("ENTER OPERATOR (+, -, *, /, //, ^): ")
        n1 = int(input("ENTER 1st NUMBER: "))
        n2 = int(input("ENTER 2nd NUMBER: "))

        if op == "+":
            print(CalcuFunction.add(n1, n2))
        elif op == "-":
            print(CalcuFunction.sub(n1, n2))
        elif op == "*":
            print(CalcuFunction.product(n1, n2))
        elif op == "/":
            print(CalcuFunction.div(n1, n2))
        elif op == "//":
            print(CalcuFunction.floor(n1, n2))
        elif op == "^":
            print(CalcuFunction.exp(n1, n2))
        else:
            print("ERROR! Invalid operator!")

    # ----------------
    # Extensions
    # ----------------
    @staticmethod
    def load(filename):
        if not filename.endswith(".py"):
            filename += ".py"

        module_name = filename[:-3]
        if CalcuFunction._extension_dir not in sys.path:
            sys.path.append(CalcuFunction._extension_dir)

        module = importlib.import_module(module_name)

        if hasattr(module, "CalcuFunction"):
            for attr in dir(module.CalcuFunction):
                if not attr.startswith("_"):
                    setattr(CalcuFunction, attr, getattr(module.CalcuFunction, attr))

File: CalcuFunction.java
import java.util.Scanner;

public final class CalcuFunction {

    private CalcuFunction() {}

    // ----------------
    // Core math
    // ----------------
    public static double add(double a, double b) { return a + b; }
    public static double sub(double a, double b) { return a - b; }
    public static double product(double a, double b) { return a * b; }
    public static double div(double a, double b) { return a / b; }
    public static double exp(double a, double b) { return Math.pow(a, b); }
    public static double floor(double a, double b) {
        return Math.floor(a / b);
    }

    // ----------------
    // CLI Calculator
    // ----------------
    public static void calculator() {
        Scanner sc = new Scanner(System.in);

        System.out.println("CalcuFunction Calculator");

        System.out.print("ENTER OPERATOR (+, -, *, /, //, ^): ");
        String op = sc.next();

        System.out.print("ENTER 1st NUMBER: ");
        double n1 = sc.nextDouble();

        System.out.print("ENTER 2nd NUMBER: ");
        double n2 = sc.nextDouble();

        switch (op) {
            case "+" -> System.out.println(add(n1, n2));
            case "-" -> System.out.println(sub(n1, n2));
            case "*" -> System.out.println(product(n1, n2));
            case "/" -> System.out.println(div(n1, n2));
            case "//" -> System.out.println(floor(n1, n2));
            case "^" -> System.out.println(exp(n1, n2));
            default -> System.out.println("ERROR! Invalid operator!");
        }
    }

  // ----------------
  // Extensions
  // ----------------

  public static Trig trig;
}
